src/yz_solrq_helper.erl

```diff
@@ -0,0 +1,225 @@
+                        {_, [notfound]} ->
+                            [{delete, yz_solr:encode_delete({bkey, BKey})} | Ops];
+                        _ ->
+                            LFPN = yz_cover:logical_partition(LI, element(1, ShortPL)),
+                            LP = yz_cover:logical_partition(LI, P),
+                            Docs = yz_doc:make_docs(Obj, Hash, ?INT_TO_BIN(LFPN), ?INT_TO_BIN(LP)),
+                            AddOps = yz_doc:adding_docs(Docs),
+                            DeleteOps = yz_kv:delete_operation(BProps, Obj, Reason, Docs, BKey, LP),
+                            %% List will be reversed, so make sure deletes happen before adds
+                            lists:append([[{add, yz_solr:encode_doc(Doc)} || Doc <- AddOps],
+                                          [{delete, yz_solr:encode_delete(DeleteOp)} || DeleteOp <- DeleteOps],
+                                          Ops])
+                    end;
+                false ->
+                    Ops
+            end
+        end, [], Entries)).
+
+send_solr_ops(Index, Ops) ->
+    try
+        T1 = os:timestamp(),
+        ok = yz_solr:index_batch(Index, Ops),
+        yz_stat:index_end(Index, length(Ops), ?YZ_TIME_ELAPSED(T1))
+    catch _:Err ->
+            yz_stat:index_fail(),
+            Trace = erlang:get_stacktrace(),
+            ?ERROR("batch for index ~s failed - ~p\n with operations: ~p : ~p",
+                   [Index, Err, Ops, Trace]),
+            yz_fuse:melt(Index),
+            {error, {Err, Trace}}
+    end.
+
+update_aae_and_repair_stats(Entries) ->
+    Repairs = lists:foldl(
+                fun({BKey, _Obj, Reason, P, ShortPL, Hash}, StatsD) ->
+                        ReasonAction = get_reason_action(Reason),
+                        Action = case ReasonAction of
+                                     delete -> delete;
+                                     _ -> {insert, Hash}
+                                 end,
+                        yz_kv:update_hashtree(Action, P, ShortPL, BKey),
+                        gather_counts({P, ShortPL, Reason}, StatsD)
+                end, dict:new(), Entries),
+    dict:map(fun({Index, IndexN}, Count) ->
+                  case Count of
+                      0 ->
+                          ok;
+                      Count ->
+                          lager:debug("Repaired ~b keys during active anti-entropy "
+                                      "exchange of partition ~p for preflist ~p",
+                                     [Count, Index, IndexN]),
+                          yz_kv:update_aae_exchange_stats(Index, IndexN, Count)
+                  end
+             end, Repairs).
+
+-spec gather_counts({p(), {p(), n()}, write_reason()}, yz_dict()) -> yz_dict().
+gather_counts({Index, IndexN, Reason}, StatsD) ->
+    case Reason of
+        {_, full_repair} ->
+            dict:update_counter({Index, IndexN}, 1, StatsD);
+        _ -> dict:update_counter({Index, IndexN}, 0, StatsD)
```

Comment:
Think this isn't needed, but leaving it in just in case there's value in seeing entries with counts of zero to know there was delete activity.  Seems unlikely.